### DISCUSSION TITLE
Concern about testing data leakage in rgcn_link_pred.py?

### DISCUSSION THREAD

**Discussion by chenshentemple** [pyg-team/pytorch_geometric#8527] [Q&A] [answered]
In the [rgcn_link_pred.py](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/rgcn_link_pred.py) example from the PyTorch Geometric repository, there is a potential issue related to edge leakage during the model's training phase. Specifically, in line 90:
`z = model.encode(data.edge_index, data.edge_type)`
the model employs all edges, encompassing those designated for validation and testing, to generate node encodings. This encoding process, detailed in lines 42-47, aggregates features from neighboring nodes using these edges. Subsequently, the resulting encoding z is utilized in positive and negative scoring within the training loop. This approach raises concerns about potential edge leakage, as the model may inadvertently access information from the validation and test sets during training. Please correct me if my understanding is inaccurate.

**Comment by EdisonLeeeee** [comment, marked as answer]
> the model employs all edges, encompassing those designated for validation and testing

Not all edges are employed during training. Instead, only the training edges are visible as the `edge_index` corresponds to the `train_edge_index`, see

https://github.com/pyg-team/pytorch_geometric/blob/44c9c1290b6ad9eeb0378df30e8397e7f5ea6383/torch_geometric/datasets/rel_link_pred_dataset.py#L111-L114

**Reply by chenshentemple** [reply]
Oh, thanks a lot. I forgot the reversed link generation and misunderstood the edge_index contains all splits because the number was greater than train. Thank you for the timely response!
